plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -0,0 +1,159 @@
+     *
+     */
+    @Override
+    public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRequest request, Executor executor) {
+        final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
+        final GetItemRequest getItemRequest = GetItemRequest
+            .builder()
+            .tableName(getTableName(request.index()))
+            .key(
+                Map
+                    .ofEntries(
+                        Map.entry(HASH_KEY, AttributeValue.builder().s(tenantId).build()),
+                        Map.entry(RANGE_KEY, AttributeValue.builder().s(request.id()).build())
+                    )
+            )
+            .build();
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
+            try {
+                final GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
+                if (getItemResponse == null || getItemResponse.item() == null || getItemResponse.item().isEmpty()) {
+                    return new GetDataObjectResponse.Builder().id(request.id()).parser(Optional.empty()).build();
+                }
+
+                String source = getItemResponse.item().get(SOURCE).s();
+                XContentParser parser = JsonXContent.jsonXContent
+                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, source);
+                return new GetDataObjectResponse.Builder().id(request.id()).parser(Optional.of(parser)).build();
+            } catch (Exception e) {
+                log.error("Exception while fetching data from DDB: " + e.getMessage(), e);
+                throw new OpenSearchException(e);
+            }
+        }), executor);
+    }
+
+    /**
+     * Deletes data document from DDB. Default tenant ID will be used if tenant ID is not specified.
+     *
+     */
+    @Override
+    public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDataObjectRequest request, Executor executor) {
+        final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
+        final DeleteItemRequest deleteItemRequest = DeleteItemRequest
+            .builder()
+            .tableName(getTableName(request.index()))
+            .key(
+                Map
+                    .ofEntries(
+                        Map.entry(HASH_KEY, AttributeValue.builder().s(tenantId).build()),
+                        Map.entry(RANGE_KEY, AttributeValue.builder().s(request.id()).build())
+                    )
+            )
+            .build();
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<DeleteDataObjectResponse>) () -> {
+            dynamoDbClient.deleteItem(deleteItemRequest);
+            return new DeleteDataObjectResponse.Builder().id(request.id()).deleted(true).build();
+        }), executor);
+    }
+
```

Comment:
Can add Update once https://github.com/opensearch-project/ml-commons/pull/2520 is merged.

Probably sould merge that and rebase this PR. CC: @dhrubo-os

Reply:
Do we need to merge both `2507` & `2520`? Seems like `2520` includes `2507` as well?

Reply:
Will add support for update and search as part of new PR.